internal/audio/ipc_common.go

```diff
@@ -143,10 +179,8 @@ func WriteIPCMessage(conn net.Conn, msg IPCMessage, pool *GenericMessagePool, dr
 	defer pool.Put(optMsg)
 
 	// Prepare header in pre-allocated buffer
-	binary.LittleEndian.PutUint32(optMsg.header[0:4], msg.GetMagic())
-	optMsg.header[4] = msg.GetType()
-	binary.LittleEndian.PutUint32(optMsg.header[5:9], msg.GetLength())
-	binary.LittleEndian.PutUint64(optMsg.header[9:17], uint64(msg.GetTimestamp()))
+	header := EncodeMessageHeader(msg.GetMagic(), msg.GetType(), msg.GetLength(), msg.GetTimestamp())
```

Comment:
Shouldn't we pass the slice of the optMsg into the encoders so we don't have to copy the result?
That way we don't need to allocate the header on line 139?